More cheat sheet includes (indexes and constraints) (#1564)
neo4j/docs-cypher: modules/cheat-sheet/pages/constraint.adoc

= link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/schema/constraints/[Constraints^]

====
include::ROOT:schema/constraints/create-constraints.adoc[tag=schema_constraints_create_property_uniqueness]

[.description]
--
Create a node property uniqueness constraint requiring `Book` nodes to have a unique `isbn` property.
An error will be thrown if an attempt is made to create the same constraint twice.
If any other node with that label is updated or created with an `isbn` property that already exists, the write operation will fail. 

Best practice is to always specify a sensible name when creating a constraint.

Property uniqueness constraints are backed by an index.

> This means that creating a property uniqueness constraint will create a range index with the same name, node label/relationship type and property combination as its owning constraint.

Reviewer: not sure, but that might be too much explanation for a cheat sheet (you be the judge here, having written the rest of it)

edit: actually might hold for more descriptions here

--
====

====

> include::ROOT:schema/constraints/create-constraints.adoc[tag=schema_constraints_create_if_not_exists]

Reviewer: this is last in the source file, while it's in second place here (at least on the basis of this PR)

Author: Yeah - but I think it makes sense to have as second. It is arguably the most important advice. Do you think it is an issue?

Reviewer: no, just something i noticed :) we don't necessarily have to apply the same order in the cheat sheet

If an `order` property of a different type is created on a relationship with that type, the write operation will fail.
====

[.neo4j-ee.aura-dbf.aura-dbp.aura-dbe.aura-dbc.aura-dsp.aura-dse]
====
include::ROOT:schema/constraints/create-constraints.adoc[tag=schema_constraints_create_property_type_dynamic_union]

[.description]
A closed dynamic union allows a node or relationship property to maintain some type flexibility whilst preventing unexpected values from being stored.
This example creates a node property type constraint requiring `tagline` properties on `Movie` nodes to be either of type `STRING` or `LIST<STRING NOT NULL>`.
====

[.neo4j-ee.aura-dbf.aura-dbp.aura-dbe.aura-dbc.aura-dsp.aura-dse]
====
include::ROOT:schema/constraints/create-constraints.adoc[tag=schema_constraints_create_key]

[.description]
--
Create a node key constraint requiring `Director` nodes to have a unique `imdbId` property as a node key.
Key constraints ensure that the property exist and the property value is unique for all nodes with a specific label or all relationships with a specific type.

> For this example, if a node with that label without `imdbId` property that is unique, the write operation will fail.

Reviewer: ```suggestion
For this example, if a node is created without the `imdbId` property being unique, the write operation will fail.
```

Author: "without a unique ..." is probably enough.

Key constraints are backed by an index.
This means that creating a key constraint will create a range index with the same name, node label/relationship type and property combination as its owning constraint.
--
====

[.neo4j-ee.aura-dbf.aura-dbp.aura-dbe.aura-dbc.aura-dsp.aura-dse]
====

> include::ROOT:schema/constraints/create-constraints.adoc[tag=schema_constraints_create_key]

Reviewer: ```suggestion
include::ROOT:schema/constraints/create-constraints.adoc[tag=schema_constraints_create_composite_key]
```
